apps/api/src/campaign-application/campaign-application.service.ts

```diff
@@ -1,24 +1,64 @@
-import { BadRequestException, HttpException, HttpStatus, Injectable } from '@nestjs/common'
+import { BadRequestException, Injectable, Logger } from '@nestjs/common'
 import { CreateCampaignApplicationDto } from './dto/create-campaign-application.dto'
 import { UpdateCampaignApplicationDto } from './dto/update-campaign-application.dto'
 import { PrismaService } from '../prisma/prisma.service'
+import { OrganizerService } from '../organizer/organizer.service'
+import { Person } from '@prisma/client'
 
 @Injectable()
 export class CampaignApplicationService {
-  constructor(private prisma: PrismaService) {}
+  constructor(private prisma: PrismaService, private organizerService: OrganizerService) {}
   async getCampaignByIdWithPersonIds(id: string): Promise<UpdateCampaignApplicationDto> {
     throw new Error('Method not implemented.')
   }
 
-  create(createCampaignApplicationDto: CreateCampaignApplicationDto) {
-    if (
-      !createCampaignApplicationDto.acceptTermsAndConditions ||
-      !createCampaignApplicationDto.transparencyTermsAccepted ||
-      !createCampaignApplicationDto.personalInformationProcessingAccepted
-    ) {
-      throw new BadRequestException('All agreements must be checked')
+  async create(createCampaignApplicationDto: CreateCampaignApplicationDto, person: Person) {
+    try {
+      if (
+        createCampaignApplicationDto.acceptTermsAndConditions === false ||
+        createCampaignApplicationDto.transparencyTermsAccepted === false ||
+        createCampaignApplicationDto.personalInformationProcessingAccepted === false
+      ) {
+        throw new BadRequestException('All agreements must be checked')
+      }
+
+      let organizer = await this.prisma.organizer.findUnique({
+        where: { personId: person.id },
+      })
+
+      if (!organizer) {
+        organizer = await this.organizerService.create({
+          personId: person.id,
+        })
+      }
+
+      const sanitizedData = {
+        campaignName: createCampaignApplicationDto.campaignName.trim(),
```

Comment:
@sashko9807 is there a sanitizing logic somewhere on the stack (nest, middleware, prisma perhaps)  that prevents Cross Site Scripting or SQL injection (i.e. my name is <script>alert('pawned')</script> or the famous [jonny drop tables](https://stackoverflow.com/questions/332365/how-does-the-sql-injection-from-the-bobby-tables-xkcd-comic-work)

Reply:
Prisma sanitizes data on insertion. Although I am not sure whether the start-end spaces are automatically trimmed.  Don't recall seeing this practice in any other part of the codebase

Will have to check for that.